src/test/java/com/palantir/docker/compose/connection/waiting/SuccessOrFailureShould.java

```diff
@@ -50,4 +50,30 @@ public void fail_from_an_exception() {
                 containsString("oh no")
             ))));
     }
+
+    @Test
+    public void succeed_on_a_lambda_that_returns_true() {
+        SuccessOrFailure successFromLambda = SuccessOrFailure.onResultOf(() -> true);
+        assertThat(successFromLambda, is(successful()));
+    }
+
+    @Test
+    public void fail_on_a_lambda_that_throws_an_exception() {
+        SuccessOrFailure failureFromLambda = SuccessOrFailure.onResultOf(() -> {
+            throw new IllegalArgumentException("oh no");
+        });
+
+        assertThat(failureFromLambda,
+                is(failureWithMessage(both(
+                        containsString("IllegalArgumentException")).and(
+                        containsString("oh no")
+                ))));
+    }
+
+    @Test
+    public void fail_on_a_lambda_that_returns_false() {
+        SuccessOrFailure failureFromLambda = SuccessOrFailure.onResultOf(() -> false);
+
+        assertThat(failureFromLambda, is(failureWithMessage("Attempt to complete healthcheck failed")));
```

Comment:
Yes, that's fair. It would be pretty trivial to add an optional string parameter which is the error message. However, in almost all cases I imagine this being used to send failures when exceptions are caught, rather than having to write this try..catch block yourself everywhere.